vault/barrier_aes_gcm.go

```diff
@@ -692,25 +722,39 @@ func (b *AESGCMBarrier) Put(ctx context.Context, entry *logical.StorageEntry) er
 
 // Get is used to fetch an entry
 func (b *AESGCMBarrier) Get(ctx context.Context, key string) (*logical.StorageEntry, error) {
+	return b.lockSwitchedGet(ctx, key, true)
+}
+
+func (b *AESGCMBarrier) lockSwitchedGet(ctx context.Context, key string, getLock bool) (*logical.StorageEntry, error) {
 	defer metrics.MeasureSince([]string{"barrier", "get"}, time.Now())
-	b.l.RLock()
+	if getLock {
+		b.l.RLock()
+	}
 	if b.sealed {
-		b.l.RUnlock()
+		if getLock {
```

Comment:
Can this `if` block be moved inside a defer func?

Reply:
It's purposefully not using defer because this is in the super hot path and defer is very slow.